packages/gestalt-datepicker/src/DateField/InternalDateField.js

```diff
@@ -0,0 +1,278 @@
+/* eslint-disable react/display-name */
+// @flow strict-local
+import { useState, forwardRef, type Node } from 'react';
+import { unstable_useDateField as useDateField } from '@mui/x-date-pickers/DateField';
+import { DatePicker as MUIDatePicker } from '@mui/x-date-pickers/DatePicker';
+import classnames from 'classnames';
+import { Box, Text, Flex, Status, TapArea, Pog } from 'gestalt';
+import styles from '../DateField.css';
+
+const ENTER: number = 13;
+const SPACE: number = 32;
+const TAB: number = 9;
+
+type CustomTextFieldProps = {|
+  disabled: boolean,
+  InputProps: {| ref: {| current: ?HTMLElement |} |},
+  focused: boolean,
+  placeholder: string,
+  value: string,
+  readOnly: boolean,
+  onClick: () => void,
+  onPaste: () => void,
+  onChange: () => void,
+  onKeyDown: () => void,
+  onMouseUp: () => void,
+  ownerState: {|
+    passthroughProps: {|
+      autoComplete: 'bday' | 'off',
+      id: string,
+      errorMessage: boolean,
+      enterKeyHint: 'enter' | 'done' | 'go' | 'next' | 'previous' | 'search' | 'send',
+      name: string,
+      onBlur: ({|
+        event: SyntheticFocusEvent<HTMLInputElement>,
+        value: string,
+      |}) => void,
+      onFocus: ({|
+        event: SyntheticFocusEvent<HTMLInputElement>,
+        value: string,
+      |}) => void,
+    |},
+  |},
+|};
+
+const CustomTextField = forwardRef(
+  (
+    {
+      disabled,
+      InputProps: { ref: containerRef } = {},
+      focused,
+      placeholder,
+      value,
+      readOnly,
+      onClick,
+      onPaste,
+      onChange,
+      onKeyDown,
+      onMouseUp,
+      ownerState,
+    }: CustomTextFieldProps,
+    inputRef,
+  ): Node => {
+    const styledClasses = classnames(
+      styles.textField,
+      styles.formElementBase,
+      styles.typographyTruncate,
+      styles.actionButton,
+      styles.layoutLarge,
+      disabled ? styles.formElementDisabled : styles.formElementEnabled,
+      ownerState?.passthroughProps?.errorMessage && !focused
+        ? styles.formElementErrored
+        : styles.formElementNormal,
+    );
+
+    return (
+      <Box ref={containerRef} rounding={4}>
+        <input
+          autoComplete={ownerState?.passthroughProps?.autoComplete ?? 'off'}
+          id={ownerState?.passthroughProps?.id}
+          className={styledClasses}
+          disabled={disabled}
+          ref={inputRef}
+          placeholder={placeholder}
+          value={value}
+          enterKeyHint={ownerState?.passthroughProps?.enterKeyHint}
+          inputMode="numeric"
+          readOnly={readOnly}
+          onClick={onClick}
+          onFocus={(event) => ownerState?.passthroughProps?.onFocus?.({ event, value })}
+          onBlur={(event) => ownerState?.passthroughProps?.onBlur?.({ event, value })}
+          onPaste={onPaste}
+          onChange={onChange}
+          onKeyDown={onKeyDown}
+          onMouseUp={onMouseUp}
+        />
+      </Box>
+    );
+  },
+);
+
+type CustomDateFieldProps = {
```

Comment:
Ideally we can make this type exact at some point (unless this is all getting thrown away)